### ISSUE TITLE
Even when selecting jobs randomly from blocks, it should be able to repeat a job immediately

### ISSUE THREAD

**Issue by smathot** [open-cogsci/omm-client#8]
Right now, if the 'Randomly select jobs from block' option is enabled in `OMMRequestJob`, then there is no way to control which job is selected next. This should be possible so that, after an error or timeout, a job can repeated immediately on the next trial.

One way (slightly hacky but functional) to accomplish this is to check if any canidate job has a variable `prioritize` with the value `1` and, if so, always select the first job that has this. If not, then a job is selected randomly from a block as currently happens.

Ping @nclaidiere

**Comment by smathot** [comment]
@nclaidiere The workaround proposed above isn't satisfactory because of the reasons we discussed before.

How about the following: when a new participant starts an experiment, an (optional) initialization script is executed that randomizes the job table for that participant. (It can also do other things, but this would be the most common scenario.) The initialization happens once, and then during the experiment the jobs are simply executed sequentially. This means that you can redo jobs by inserting them as new jobs at a specific position, or by resetting their state to pending (in which case they're redone immediately).

Ideally, this initialization script would be executed when the participants starts the experiment for the first time. But it may cause a bit of a delay because it involves a lot of back-and-forths with the server (I have to time this), in which case it might be better to explicitly initialize the experiment yourself with a separate script.

Does this sound like a suitable approach?

**Comment by nclaidiere** [comment]
@smathot That sounds great, we discussed it with Joel, let me rephrase what you proposed to see if we understood correctly:
- by default the program will go through the jobs in the order of Pending jobs in the job list, no randomisation
- an initialisation script can be inserted at the beginning of the experiment that will randomise jobs within blocks just once at the beginning of the experiment
- in this context if jobs are set back to pending they will be represented in the order of the randomised block, so if one job is set back to pending it is represented immediately

Correct?

The only problem is that blocks are often performed several times (e.g. to reach criteria) and they need to be randomised every time they are represented. So would it be possible to do that:
- an initialisation script can be inserted at the beginning of the experiment that will randomise jobs within blocks _at the beginning of every block_? The order will then remain fixed until the end of block. 

What do you think?

**Comment by smathot** [comment]
After giving it some more thought, I think it's best to implement this partially on the server through a new end-point for accessing data. That will make everything on the client side much easier and more flexible. open-cogsci/omm-server#95
